chore: merge with dsp ingest

siers opened the pull request.

seakayone commented on .github/workflows/build-and-test.yml:
```diff
@@ -97,6 +97,22 @@ jobs:
+          java-version: 21
+          cache: sbt
+      - name: Run integration tests
+        run: export DOCKER_BUILDKIT=1; ./sbtx Docker/publishLocal; ./sbtx ingestIntegration/test
```
For the other e2e/it tests we have a job which parses the xunit xml report, can you please add this too?

seakayone commented:
Here are some more notes on the dsp-ingest merge:

1. Building the docker containers with `make`
    - Until now all docker containers could be built with a single command `make  docker-build`
      You have added the `docker-publish-ingest-image` target for the publishing and included this into the `docker-publish`.
      This is good and should be kept, but it does not allow to only build the ingest image locally.
      Building only the ingest image should be available with `docker-build-ingest-image` target.
      The `docker-build` target should include `docker-build-ingest-image`. 
2. Executing the ingest integration tests with `just`
   - Current `just` has the recipes `test`, `test-e2e` and `test-it`. 
     For sake of consistency we should add recipes `test-ingest` and `test-integration-ingest` to execute the unit and integration tests of the `ingest` module.
3. Running the ingest integration test locally fails initially
    - The error is 
      ```scala
        com.github.dockerjava.api.exception.NotFoundException: Status 404: {"message":"failed to resolve reference \"docker.io/daschswiss/dsp-ingest:v31.21.2-14-g4ba66bd-feat-merge-with-dsp-ingest\": docker.io/daschswiss/dsp-ingest:v31.21.2-14-g4ba66bd-feat-merge-with-dsp-ingest: not found"}
      ```
      This is because the image is not built locally yet.
      I need to build the image first with `sbt injest/Docker/publishLocal` before I can run the integration tests.
      This should be either documented of fixed (AFAIR in the api integration tests this was fixed during my last leave)
4. Ensure building the ingest image is alwasys using the current knora-sipi base image
    - The ingest image depends on the base image `knora-sipi`.
      `knora-sipi` is built as part of the `docker-build` target.
       Both image are build in the same repository and released together.
       We should remove the hardcoded version `knoraSipiVersion` in `Dependencies.scala` and use the current release version instead.
       This way we ensure that both images are always in sync and we avoid unnecessary manual version bumps.
5. Documentation
    - The `README.md` should be updated to reflect the changes.
      It should include instructions on how to build and run the ingest image, as well as how to execute the tests.
      Additionally, any relevant information about the merge and its implications should be documented for future reference (`CHANGELOG.md`).

siers commented:
@seakayone 

1. `✔️` Building the docker containers with `make`
    I left this out, because I wasn't sure anyone is using these really, but they're for dependencies in the Makefile in the workflow. Added.

2. `✔️` Executing the ingest integration tests with `just`
    These definitions are empty wrappers, I see little use, but ok, added.

3. `~` Running the ingest integration test locally fails initially
    I am not sure how this is solved, since I couldn't find any occurences publish in scala source over the project.
    If you are thinking of running this through makefile, this is not a fix really, since you can always run the command without the docker build and the problem will still be there, but the makefile command is now there, since it was customary for the CI process.

4. `✔️` Ensure building the ingest image is alwasys using the current knora-sipi base image
    This required some tinkering, but I got this to work.

5. `-` Documentation
    If you don't mind, I want to check how far the builds work and I can promise to add this in the next pull request.

siers commented on .github/workflows/build-and-test.yml:
```diff
@@ -19,7 +19,7 @@ jobs:
           java-version: ${{ env.JAVA_VERSION }}
       - name: Run all tests
-        run: ./sbtx -v coverage "webapi/test" coverageAggregate
+        run: ./sbtx -v coverage webapi/test ingest/test coverageAggregate
```
Fixed with the second attempt.
